rosidl_parser/rosidl_parser/__init__.py

```diff
@@ -445,8 +445,7 @@ def parse_value_string(type_, value_string):
     if type_.is_primitive_type() and not type_.is_array:
         return parse_primitive_value_string(type_, value_string)
 
-    # TODO(mikaelarguedas) change this condition once string escape function is implemented
-    if type_.is_primitive_type() and type_.is_array and type_.type != 'string':
+    if type_.is_primitive_type() and type_.is_array:
```

Comment:
if we allow to not quote strings it is valid to declare:
`string name a, b`
this results in a string containing `"a, b"` and is totally valid.

Using the same logic for arrays it's valid to declare:
`string[] names [a, b, c, d]`
in that case we don't know if it's a single element containing `"a, b, c, d" or four element separated by commas or any other combination.
That's what I meant by making the quoting mandatory. Otherwise we'd need to define a delimiter that is not allowed in strings which doesn't make much sense.